### PR TITLE
Fix loading for Rails 4.2

### PR DESCRIPTION
**Pull request by james2m**
I added some updates to make observers work correctly with Rails 4.2.
Bumped the Rspec version.
Added Ruby 2.2 support.
Re-structured the files to a more standard structure.

### CODE REVIEW

**Comment by james2m** [comment]
Fixed test environment for compatibility with state_machines-activemodel dependencies.

File: state_machines-activemodel-observers.gemspec [hunk 1]
```diff
@@ -17,13 +17,15 @@ Gem::Specification.new do |spec|
   spec.executables   = spec.files.grep(%r{^bin/}) { |f| File.basename(f) }
   spec.test_files    = spec.files.grep(%r{^spec/})
   spec.require_paths = ['lib']
-  spec.required_ruby_version     = '>= 1.9.3'
+  spec.required_ruby_version     = '>= 2.0.0'
 
-  spec.add_dependency 'state_machines-activemodel'    , '>= 0.0.2'
+  spec.add_dependency 'state_machines', '>= 0.1.0'
+  spec.add_dependency 'state_machines-activemodel', '>= 0.0.3'
+  spec.add_dependency 'activemodel', '~> 4.0'
```
Comment (seuros) [inline state_machines-activemodel-observers.gemspec:24]: This won't work. The activemodel integration depend on 4.1+.

File: state_machines-activemodel-observers.gemspec [hunk 2]
```diff
@@ -17,13 +17,15 @@ Gem::Specification.new do |spec|
   spec.executables   = spec.files.grep(%r{^bin/}) { |f| File.basename(f) }
   spec.test_files    = spec.files.grep(%r{^spec/})
   spec.require_paths = ['lib']
-  spec.required_ruby_version     = '>= 1.9.3'
+  spec.required_ruby_version     = '>= 2.0.0'
```
Comment (seuros) [inline state_machines-activemodel-observers.gemspec:20]: The jruby users are still in the 1.9.3 for now.

File: state_machines-activemodel-observers.gemspec [hunk 3]
```diff
@@ -17,13 +17,15 @@ Gem::Specification.new do |spec|
...
+  spec.add_dependency 'state_machines', '>= 0.1.0'
+  spec.add_dependency 'state_machines-activemodel', '>= 0.0.3'
+  spec.add_dependency 'activemodel', '~> 4.0'
   spec.add_dependency 'rails-observers'
 
   spec.add_development_dependency 'bundler', '>= 1.6'
   spec.add_development_dependency 'rake', '>= 10'
   spec.add_development_dependency 'appraisal', '>= 1'
-  spec.add_development_dependency 'rspec' , '3.0.0.beta2'
+  spec.add_development_dependency 'rspec' , '~>3.1.0'
```
Comment (seuros) [inline state_machines-activemodel-observers.gemspec:30]: :+1:

**Pull request closed without merging**